runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsPlugin.java

```diff
@@ -81,12 +92,56 @@
 
 	private static final Set<Integer> BARROWS_LADDERS = Sets.newHashSet(NullObjectID.NULL_20675, NullObjectID.NULL_20676, NullObjectID.NULL_20677);
 
+	private static final int TUNNEL_WIDGET_GROUP_ID = 229;
+	private static final String TUNNEL_MESSAGE = "You've found a hidden tunnel, do you want to enter?";
+
 	@Getter(AccessLevel.PACKAGE)
 	private final Set<WallObject> walls = new HashSet<>();
 
 	@Getter(AccessLevel.PACKAGE)
 	private final Set<GameObject> ladders = new HashSet<>();
 
+	@Getter(AccessLevel.PACKAGE)
+	@Nullable
+	private BarrowsBrothers tunnelBrother;
+
+	private static final Map<BarrowsBrothers, WorldArea> BARROWS_ROOMS = new HashMap<BarrowsBrothers, WorldArea>()
```

Comment:
@raiyni suggested I saved these areas as simple rectangles. But I felt like it was more appropriate to use existing classes that define the world area. What is being done in other plugins?